### PR TITLE
Show feedback icon option on parts. Fixes #417

### LINKED ISSUE #417: "Show feedback icon" option on parts
Parts should have an option dictating whether to display the tick/cross feedback icon.
I'm writing a gapfill with a custom marking script, and since the mark awarded depends on the combination of two gapfills, it doesn't make any sense to show individual feedback for each gapfill.

### PR DESCRIPTION
**Pull request by janga1997**
I am not too sure about the third change, in the display-base.js file.

### CODE REVIEW

File: themes/default/files/scripts/display-base.js [hunk 1]
```diff
@@ -273,6 +273,11 @@ Numbas.queueScript('display-base',['controls','math','xml','util','timing','jme'
                 }
             }),
             message: ko.computed(function() {
+
+                if (settings.showFeedbackIcon) {
```
Comment (christianp) [inline themes/default/files/scripts/display-base.js]: This is in the wrong place, and does the opposite of what it should!
You want to return `invisible` when `settings.showFeedbackIcon` is `false`, i.e. `if(!settings.showFeedbackIcon)`.

This test should be at the top of the `iconClass` observable, not here in `message`.

**Comment by christianp** [comment]
The code to load the setting is fine, but you're using it in the wrong place. You're also not passing it through to `showScoreFeedback`: if you look at [line 202 of `themes/default/files/scripts/part-display.js`](https://github.com/numbas/Numbas/blob/946b7a0c53eb935e18e01eb897670ddd3cca540e/themes/default/files/scripts/part-display.js#L202), the exam's settings object is being passed through. You need to make a copy and add in the `showFeedbackIcon` option:

```javascript
var feedback_settings = Numbas.util.copyobj(p.question.exam.settings);
feedback_settings.showFeedbackIcon = p.settings.showFeedbackIcon;
this.scoreFeedback = display.showScoreFeedback(this, feedback_settings);
```

**Comment by christianp** [comment]
It works! Thanks!
I rewrote the commit message, which has made GitHub lose track of where your commit went, but it is merged.

**Comment by janga1997** [comment]
I am sorry, but did you mean to close this or merge this?

**Comment by christianp** [comment]
I clsoed it - as I said, since I rewrote your commit message, merging on GitHub would have made a mess of the commit history.

**Comment by janga1997** [comment]
@christianp Oh alright. I think I almost understand you.
Thanks for all the help.

**Pull request closed without merging**